src/utrfx/util.py

```
@@ -8,35 +8,53 @@
 
 def download_fasta_from_ensembl(transcript_id: str, timeout: float = 30.,) -> str:
     """
-    Download a FASTA file containing the cDNA sequence for a given transcript from Ensembl's REST API and
+    Download a FASTA file containing the cDNA sequence (spliced mRNA) for a given transcript from Ensembl's REST API and
     return only the nucleotide sequence (without the FASTA header).
 
+    If the transcript is in the negative strand, it will be given the reverse complement of the spliced mRNA found in Ensembl's REST API,
+    which is it located in the positive strand.         
+
     :param transcript_id: Ensembl transcript identifier e.g. `ENST00000381418`
     """
-    base_url = f"https://rest.ensembl.org/sequence/id/{transcript_id}?type=cdna&content-type=text/x-fasta"
+    base_url_cdna = f"https://rest.ensembl.org/sequence/id/{transcript_id}?type=cdna&content-type=text/x-fasta"
     
-    response = requests.get(base_url, timeout=timeout)
+    response = requests.get(base_url_cdna, timeout=timeout)
 
     if response.status_code == 200:
         lines = response.text.splitlines()
-        return ''.join(lines[1:])
+
+        cdna_sequence = ''.join(lines[1:])
+
+        base_url_strand = f"https://rest.ensembl.org/lookup/id/{transcript_id}?content-type=application/json"
+
+        response_strand = requests.get(base_url_strand, timeout=timeout)
+        data = response_strand.json()
+
+        strand = data.get("strand", None)
+        
+        if strand > 0: 
+            return cdna_sequence
+        elif strand < 0:
+            return cdna_sequence.translate(str.maketrans("ATCG", "TAGC"))[::-1]
+        else:
+            return "Strand not found."
```

You cannot just return an arbitrary `str` if you promise a nucleotide sequence in the signature. Absence of strand is abnormal, and therefore a case for an exception.

Please raise an exception if you cannot obtain the strand.